internal/plugin/process_manager.go

```diff
@@ -0,0 +1,117 @@
+package plugin
+
+import (
+	"fmt"
+	"log"
+	"os/exec"
+	"syscall"
+	"time"
+)
+
+// TODO: this can probably be defaulted to this, but overwritten on a per-plugin basis
+const GRACEFUL_SHUTDOWN_DELAY = 30 * time.Second
+const MAX_RESTART_BACKOFF = 30 * time.Second
```

Comment:
I think these could probably be combined into one `const` block, like this:
```suggestion
const (
	GRACEFUL_SHUTDOWN_DELAY = 30 * time.Second
	MAX_RESTART_BACKOFF = 30 * time.Second
)
```

Effective Go and the Go stdlib also usually prefers lowerCamelCase/UpperCamelCase (depending on whether exported) for constant names. E.g., per https://go.dev/wiki/CodeReviewComments:
> See https://go.dev/doc/effective_go#mixed-caps. This applies even when it breaks conventions in other languages. For example an unexported constant is maxLength not MaxLength or MAX_LENGTH.